spec/classes/init_facterdb_spec.rb

```diff
@@ -0,0 +1,395 @@
+require 'spec_helper'
+describe 'ssh' do
+  debian = {
+    supported_os: [
+      {
+        'operatingsystem'        => 'Debian',
+        'operatingsystemrelease' => ['9'],
+      },
+    ],
+  }
+
+  header = <<-END.gsub(%r{^\s+\|}, '')
+    |# This file is being maintained by Puppet.
+    |# DO NOT EDIT
+    |#
+    |# See https://man.openbsd.org/sshd_config for more info
+    |
+    |Host *
+  END
+
+  rh_default_content = <<-END.gsub(%r{^\s+\|}, '')
+    |  ForwardX11Trusted yes
+    |  GSSAPIAuthentication yes
+    |  SendEnv LANG
+    |  SendEnv LANGUAGE
+    |  SendEnv LC_ADDRESS
+    |  SendEnv LC_ALL
+    |  SendEnv LC_COLLATE
+    |  SendEnv LC_CTYPE
+    |  SendEnv LC_IDENTIFICATION
+    |  SendEnv LC_MEASUREMENT
+    |  SendEnv LC_MESSAGES
+    |  SendEnv LC_MONETARY
+    |  SendEnv LC_NAME
+    |  SendEnv LC_NUMERIC
+    |  SendEnv LC_PAPER
+    |  SendEnv LC_TELEPHONE
+    |  SendEnv LC_TIME
+    |  SendEnv XMODIFIERS
+  END
+
+  on_supported_os.sort.each do |os, os_facts|
+    context "on #{os} with default values for parameters" do
+      let(:facts) { os_facts }
+
+      it { is_expected.to compile.with_all_deps }
+      it { is_expected.to contain_class('ssh') }
+
+      it do
+        is_expected.to contain_package('openssh-clients').only_with(
+          {
+            'ensure'    => 'installed',
+            'source'    => nil,
+            'adminfile' => nil,
+          },
+        )
+      end
+
+      content = if os_facts[:os]['family'] == 'RedHat'
+                  rh_default_content
+                else
+                  ''
+                end
+
+      it do
+        is_expected.to contain_file('ssh_config').only_with(
+          {
+            'ensure'  => 'file',
+            'path'    => '/etc/ssh/ssh_config',
+            'owner'   => 'root',
+            'group'   => 'root',
+            'mode'    => '0644',
+            'content' => header + content,
+            'require' => 'Package[openssh-clients]',
+          },
+        )
+      end
+
+      it { is_expected.not_to contain_exec("mkdir_p-#{os_facts[:root_home]}/.ssh") }
+      it { is_expected.not_to contain_file('root_ssh_dir') }
+      it { is_expected.not_to contain_file('root_ssh_config') }
+
+      it { is_expected.to have_sshkey_resource_count(0) }
+
+      it do
+        is_expected.to contain_file('ssh_known_hosts').only_with(
+          {
+            'ensure'  => 'file',
+            'path'    => '/etc/ssh/ssh_known_hosts',
+            'owner'   => 'root',
+            'group'   => 'root',
+            'mode'    => '0644',
+            'require' => 'Package[openssh-clients]',
+          },
+        )
+      end
+
+      # FIXME: add test for
+      # Sshkey <<||>> {
```

Comment:
We should just leave this collection functionality out and document in the readme and let people implement in a profile.

Reply:
Do you mean that we should remove the functionality from the manifest ?
https://github.com/ghoneycutt/puppet-module-ssh/blob/v4/manifests/init.pp#L670-L675

Doesn't it makes sense to completly remove both functionalities ($ssh_key_import & $key_export) then ?

Reply:
Yep! There have always been issues around these and just showing people how to do it would be better.

Reply:
added a check box for adding the documentation in the main PR https://github.com/ghoneycutt/puppet-module-ssh/pull/307

Will remove both functionalities eventualy.